### PR TITLE
updated documentation for Embedding layer

### LINKED ISSUE #4682: Update documentation for Embedding layer
The documentation corresponding [torch.nn.Embedding](http://pytorch.org/docs/master/nn.html) mentions that ```Keep in mind that only a limited number of optimizers support sparse gradients: currently it’s optim.SGD (cuda and cpu), and optim.Adagrad (cpu)```. This is outdated and now `SparseAdam` is also supported.

### PR DESCRIPTION
**Pull request by shagunsodhani**
The documentation corresponding torch.nn.Embedding mentions that Keep in mind that only a limited number of optimizers support sparse gradients: currently it’s optim.SGD (cuda and cpu), and optim.Adagrad (cpu). This is outdated and now SparseAdam is also supported. Updated the document to reflect the same. This fixes #4682

### CODE REVIEW

**Comment by soumith** [comment]
thanks Shagun!
